lib/routers/profile/pil.js

```diff
@@ -86,34 +86,38 @@ const attachEstablishmentDetails = (req, res, next) => {
     .catch(next);
 };
 
-router.param('pilId', (req, res, next, id) => {
+router.param('pilId', async (req, res, next, id) => {
   if (!isUUID(id)) {
     return next(new NotFoundError());
   }
 
-  const { PIL } = req.models;
+  const { PIL, Profile } = req.models;
   const { withDeleted } = req.query;
   const queryType = withDeleted ? 'queryWithDeleted' : 'query';
 
-  Promise.resolve()
-    .then(() => {
-      return PIL[queryType]().findById(id)
-        .where(builder => {
-          if (req.profileId) {
-            return builder.where({ profileId: req.profileId });
-          }
-        });
-    })
-    .then(pil => {
-      if (!pil) {
-        throw new NotFoundError();
+  const pil = await PIL[queryType]()
+    .findById(id)
+    .where(builder => {
+      if (req.profileId) {
+        return builder.where({ profileId: req.profileId });
       }
-      pil.procedures = pil.procedures || [];
-      pil.species = pil.species || [];
-      req.pil = pil;
-      next();
-    })
-    .catch(next);
+    });
+
+  if (!pil) {
+    return next(new NotFoundError());
+  }
+
+  const profile = await Profile.query().findById(pil.profileId);
+
+  if (!pil.licenceNumber || pil.licenceNumber === profile.pilLicenceNumber) {
+    pil.licenceNumber = profile.pilLicenceNumber;
```

Comment:
do we need the 2nd condition here? Isn't it just a noop?

Reply:
no, this is addressing Lenny's earlier comment: We always want to use the licence number from the profile, apart from if it is different (if you are looking at a legacy migrated pil)